rfcs/text/0011_global_search.md

```diff
@@ -0,0 +1,709 @@
+using AST to find all usages to `application.register` inspect the parameters, and generates a server file 
+containing the applications. The server-side `application` result provider would then just read this file and uses it 
+to return application results.
+
+However
+- At the parsing would be done at build time, we would not be able to generate entries for any 3rd party plugins
+- As entries for every existing applications would be generated, the search provider would to be able to know which
+applications are actually enabled/accessible at runtime to filter them, which is all but easy
+- It will also not contains test plugin apps, making it really hard to FTR
+- AST parsing is a complex mechanism for an already unsatisfactory alternative
+ 
+### Duplicated server-side `application.register` API 
+ 
+One other option would be to duplicate the `application.register` API on the server side, with a subset of the 
+client-side metadata.
+
+```ts
+core.application.register({
+  id: 'app_status',
+  title: 'App Status',
+  euiIconType: 'snowflake'
+});
+```
+
+This way, the applications could be searchable from the server using this server-side `applications` registry.
+
+However
+- It forces plugin developers to add this API call. In addition to be a very poor developer experience, it can also
+  very easily be forgotten, making a given app non searchable
+- client-side only plugins would need to add a server-side part to their plugin just to register their application on
+  the server side
+  
+## `GlobalSearchResult.url` could be a struct instead of a url for internal results.
+
+The initial idea for `GlobalSearchResult.url` was to have a polymorphic structure instead of just a string:
+
+```ts
+url: { absUrl: string } | { application: string; path?: string };
+```
+
+That was making it way easier to redirect to an internal result from the UI, as we could directly call
+`application.navigateToApp(application, { path })`.
+
+However, that didn't bring answer for the (future) need to be able to search for and redirect to object in 
+different spaces. We could then have changed the structure to
+
+```ts
+url: { absUrl: string } | { application: string; path?: string, space?: string };
+```
+
+But this had some caveats:
+- `space` is an xpack plugin, adding this `space` property in an oss part of the code is problematic
+- The `space` API is not callable from core or oss, meaning that we would have to 'forge' the url to this space anyway
+- this is really not generic. If another plugin was to alter the basepath in another way, we would have needed to add it another property 
+
+So even if the 'parsable absolute url' approach seems fragile, it still felt better than this alternative.
+
+## The GlobalSearch service could be provided as a plugin instead of a core service
+
+The GlobalSearch API could be provided and exposed from a plugin instead of `core`.
+
+Pros:
```

Comment:
Opening a thread for the core vs plugin discussion

Reply:
Unless someone has some pros or cons to add, it's currently planned to change the RFC to be providing the GS API as a plugin instead of a core service.

The leading argument for that being that the global search bar (name YTBD) is probably going to be a licensed feature, meaning that having it in core would not be an option.

Reply:
done in https://github.com/elastic/kibana/pull/64284/commits/8180c3af0369acbc4b29b1809bf3f72f2093d568